src/nss-resolve/nss-resolve.c

```diff
@@ -464,7 +464,7 @@ enum nss_status _nss_resolve_gethostbyname3_r(
                 if (r < 0)
                         goto fail;
 
-                if (q.family != af)
+                if (!IN_SET(q.family, AF_INET, AF_INET6))
```

Comment:
Rather than modifying this, please fix the condition in the first loop.

Reply:
af argument needs to be specified (when unspecified, AF_INET is assumed). Hence, the method cannot provide both IPv4 and IPv6 addresses simultaneously.